### PR TITLE
Add named run lists attribute policy revisions detail

### PR DESCRIPTION
**Pull request by vsingh-msys**
## Description

 - Add `named_run_lists` attributes to `GetPolicyRevisionDetail`
 - Updated Included policy locks struct.
 - Minor updates
 - Cover minor test cases.

## Related Issue

NA
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: policy.go [hunk 1]
```diff
@@ -25,14 +24,23 @@ type PolicyRevisionDetail map[string]interface{}
 // RevisionDetailsResponse is returned from the chef-server for Get Requests to /policies/<policy-name>/revisions/<rev-id>
 type RevisionDetailsResponse struct {
 	RevisionID           string                  `json:"revision_id,omitempty"`
-	Name                 string                  `json:"name",omitempty`
-	RunList              []string                `json:"run_list,omitempty`
```
Comment (vsingh-msys) [inline policy.go:29]: Not populating the `run_list` due to double-quote was missing.

File: policy.go [hunk 2]
```diff
@@ -25,14 +24,23 @@ type PolicyRevisionDetail map[string]interface{}
...
-	IncludePolicyLocks   []string                `json:"included_policy_locks,omitempty"`
+	Name                 string                  `json:"name,omitempty"`
+	RunList              []string                `json:"run_list,omitempty"`
+	NamedRunList         map[string][]string     `json:"named_run_lists,omitempty"`
+	IncludedPolicyLocks  []IncludedPolicyLocks   `json:"included_policy_locks,omitempty"`
```
Comment (vsingh-msys) [inline policy.go:30]: IncludedPolicyLocks returns the map of name and policy detail.

eg.
```
included_policy_locks": [
    {
      "name": "policy1",
      "revision_id": "7b40995ad1150ec56950c757872d6732aa00e76382dfcd2fddeb3a971e57ba9c",
      "source_options": {
        "path": "./Policyfile.lock.json"
      }
    }
  ],

```

**Comment by vsingh-msys** [comment]
@tyler-ball & @marcparadise any comments??

File: policy_test.go [hunk 3]
```diff
@@ -100,6 +100,24 @@ func TestGetPolicyRevision(t *testing.T) {
...
+	if data.RunList[0] != "recipe[base::default]" {
+		t.Errorf("Unexpected policy run list: %+v", data.RevisionID)
+	}
+
+	if val, ok := data.NamedRunList["os"]; !ok {
+		t.Error("Expected os NamedRunList policy to be present in the policy information")
+	} else if val[0] != "recipe[hardening::default]" {
+		t.Error("Expected named run list for the policy, got: ", val[0])
+	}
+
```
Comment (tyler-ball) [inline policy_test.go:120]: On closer inspection - can you add validation for the `included_policy_locks` field being parsed?

**Pull request merged**